src/use_printer.rs

```diff
@@ -1,46 +1,26 @@
 use std::{
     cell::{OnceCell, RefCell},
+    ptr,
     rc::Rc,
 };
 
 use termcolor::{Buffer, BufferWriter};
 
-use crate::{
-    args::OutputMode,
-    printer::{Standard, StandardBuilder},
-};
+use crate::{printer::Standard, Args};
 
-type Printer = Standard<Buffer>;
+pub type Printer = Standard<Buffer>;
 
 thread_local! {
-    static PRINTER: OnceCell<(Rc<RefCell<Printer>>, OutputMode)> = Default::default();
+    static PRINTER: OnceCell<(Rc<RefCell<Printer>>, *const Args)> = Default::default();
```

Comment:
Look at me being clever and avoiding lifetime stuff by storing a `*const Args` here (to maintain the same "invariant sanity-check" as was here previously) and knowing that I won't run into `unsafe` because I'm not actually going to dereference that pointer (just compare it by pointer identity)